okhttp/src/test/java/okhttp3/internal/concurrent/TaskRunnerRealBackendTest.kt

```diff
@@ -82,11 +82,22 @@ class TaskRunnerRealBackendTest {
       return@schedule -1L
     }
 
-    queue.awaitIdle(TimeUnit.MILLISECONDS.toNanos(500))
+    queue.idleLatch().await(500, TimeUnit.MILLISECONDS)
 
     assertThat(log.take()).isEqualTo("failing task running")
     assertThat(log.take()).isEqualTo("uncaught exception: java.lang.RuntimeException: boom!")
     assertThat(log.take()).isEqualTo("normal task running")
     assertThat(log).isEmpty()
   }
+
+  @Test fun idleLatchAfterShutdown() {
+    queue.schedule("task") {
+      Thread.sleep(250)
```

Comment:
nit: on it's own 250ms is fine.  As part of a test suite 40 tests taking this approach add 10 seconds to running the tests and is still more flaky then using e.g. CountDownLatch here.  

I've raised https://github.com/square/okhttp/issues/5673 to fix such usages.